src/main/java/hu/oe/nik/szfmv/automatedcar/powertrain/Transmission.java

```diff
@@ -0,0 +1,114 @@
+package hu.oe.nik.szfmv.automatedcar.powertrain;
+
+public class Transmission implements ITransmission {
+
+    /**
+     * IMPORTANT!!!
+     * These constant values define the gear shifting thresholds in {@link CarTransmissionMode#D_DRIVE D} mode.
+     * Changing these values can effect the workings of the car extremely.
+     * Tweak carefully.
+     * TODO: calibrate !!!
+     */
+    private final int RPM_gear_0_upshift_threshold = 2000;
+    private final int RPM_gear_1_downshift_threshold = 1500;
+    private final int RPM_gear_1_upshift_threshold = 6000;
+    private final int RPM_gear_2_downshift_threshold = 3000;
+    private final int RPM_gear_2_upshift_threshold = 6000;
+    private final int RPM_gear_3_downshift_threshold = 3000;
+    private final int RPM_gear_3_upshift_threshold = 6000;
+    private final int RPM_gear_4_downshift_threshold = 3000;
+    private final int RPM_gear_4_upshift_threshold = 6000;
+    private final int RPM_gear_5_downshift_threshold = 3000;
+
+    /**
+     * Field storing the automatic transmissions internal gear state in {@link CarTransmissionMode#D_DRIVE D} mode.
+     * Zero equals neutral.
+     * Range: 0-5
+     */
+    private int drive_internal_gear = 0;
+
+    /**
+     * Field storing the transmission's actual gear.
+     */
+    private CarTransmissionMode gearMode = CarTransmissionMode.P_PARKING;
+
+    /**
+     * Getter method for the {@link Transmission#drive_internal_gear} field.
+     * @return Returns the actual gear. 
+     */
+    public int getDrive_internal_gear(){
```

Comment:
The Java naming convention uses simple PascalCase (and pascalCase) and SCREAMING_SNAKE_CASE for static finals (constants). Avoid using underscore for non-constant members in production code (especially for exposed API, arguably usable for private things sometimes).
This also applies to private fields.